src/inmanta/module.py

```diff
@@ -1988,10 +1990,13 @@ def verify_python_requires(self) -> None:
         """
         Verifies no incompatibilities exist within the Python environment with respect to installed module v2 requirements.
         """
-        if not env.ActiveEnv.check(in_scope=re.compile(f"{ModuleV2.PKG_NAME_PREFIX}.*")):
-            raise CompilerException(
-                "Not all installed modules are compatible: requirements conflicts were found. Please resolve any conflicts"
-                " before attempting another compile. Run `pip check` to check for any incompatibilities."
+        constraints: List[Requirement] = [Requirement.parse(item) for item in self.collect_python_requirements()]
+        constraints_for_marker = filter(lambda c: not (c.marker and not c.marker.evaluate()), constraints)
```

Comment:
I find this a easier to read:
```suggestion
        constraints_for_marker = filter(lambda c: not c.marker or c.marker.evaluate(), constraints)
```